test/test_mahalanobis_mixin.py

```diff
@@ -603,6 +603,45 @@ def test_singular_covariance_init_or_prior(estimator, build_dataset):
       assert str(raised_err.value) == msg
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize('estimator, build_dataset',
+                         [(ml, bd) for idml, (ml, bd)
+                          in zip(ids_metric_learners,
+                                 metric_learners)
+                          if idml[:3] in ['MMC']],
+                         ids=[idml for idml, (ml, _)
+                              in zip(ids_metric_learners,
+                                     metric_learners)
+                              if idml[:3] in ['MMC']])
+def test_singular_covariance_init_of_non_strict_pd(estimator, build_dataset):
+    """Tests that when using the 'covariance' init or prior, it returns the
+    appropriate warning if the covariance matrix is singular, for algorithms
+    that don't need a strictly PD init. Also checks that the returned
+    inverse matrix has finite values
```

Comment:
This is not what you check, right? You seem to check the learned `M`.
Any reason you do this and not a call to `_initialize_metric_mahalanobis` like in the other test? Maybe you could check both things in both tests?

Reply:
Yes, you are totally right! Forgot to set the max iteration in order to make it stop before the first iteration.

I called `_initialize_metric_mahalanobis` directly in the other test because currently there is no algorithm that could reproduce that setting. But setting `max_iter = 0` should keep it from iterating and would result in the `M` from the initialization.